src/librustc_trans/back/linker.rs

```diff
@@ -101,6 +113,244 @@ pub trait Linker {
     fn subsystem(&mut self, subsystem: &str);
 }
 
+pub struct LdLinker<'a> {
+    cmd: &'a mut Command,
+    sess: &'a Session,
+    info: &'a LinkerInfo,
+}
+
+impl<'a> LdLinker<'a> {
+    fn takes_hints(&self) -> bool {
+        !self.sess.target.target.options.is_like_osx
+    }
+}
+
+impl<'a> Linker for LdLinker<'a> {
```

Comment:
Most of this seems duplicated from a definition above, would it be possible to share code instead of duplicating?